paramgenerator/generateparamsbi.py

```diff
@@ -218,22 +240,25 @@ def serialize_q15(countries):
    for country, count in countries:
       writer.append([country], [count])
 
-def serialize_q16(persons, tagclasses, countries):
-   writer = ParamsWriter("q16", ["person","tag","country"])
+def serialize_q16(persons, tagclasses, countries, path_bounds):
+   writer = ParamsWriter("q16", ["person","tag","country","minPathDistance","maxPathDistance"])
    random.seed(1988+2)
    for tag, count_a in tagclasses:
       for country, count_b in countries:
-         writer.append([str(persons[random.randint(0,len(persons))]), tag, country], [0, count_a, count_b])
+         for minDist, maxDist in path_bounds:
+            writer.append([str(persons[random.randint(0,len(persons))]), tag, country, str(minDist), str(maxDist)], [0, count_a, count_b, 0, 0])
 
 def serialize_q17(countries):
    writer = ParamsWriter("q17", ["country"])
    for country, count in countries:
       writer.append([country], [count])
 
-def serialize_q18(post_weeks):
-   writer = ParamsWriter("q18", ["creationDate"])
+def serialize_q18(post_weeks, lengths, languages):
+   writer = ParamsWriter("q18", ["creationDate", "lengthThreshold", "languageCodes"])
```

Comment:
why `languageCodes` instead of `languages`? spec says `String[] languages` @ https://ldbc.github.io/ldbc_snb_docs_snapshot/ldbc-snb-specification.pdf#page=65